### PR TITLE
[Programmes-6929] structured data for collections

### PR DESCRIPTION
**Pull request by tomwinnington**
Ticket:
https://jira.dev.bbc.co.uk/browse/PROGRAMMES-6929

Link:
http://sandbox.bbc.co.uk/programmes/p01drwny/episodes/downloads

Testing tool:
https://search.google.com/structured-data/testing-tool/u/0/

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/Controller/Helpers/StructuredDataHelper.php [hunk 1]
```diff
@@ -124,15 +125,16 @@ public function getSchemaForProgrammeContainerAndParents(Programme $programmeCon
...
+    public function getSchemaForCollection(Collection $collection, Programme $programme): array
     {
-        $schemaContext = $this->getSchemaForCollectionContainer($programmeContainer);
-        if ($programmeContainer->isTlec()) {
+
+        $schemaContext = $this->schemaHelper->getSchemaForCollection($collection);
+        if ($collection->isTlec() or !$programme) {
             return $schemaContext;
         }
-        $ancestry = $programmeContainer->getAncestry();
```
Comment (BjornTwachtmann) [inline src/Controller/Helpers/StructuredDataHelper.php]: getSchemaForProgrammeContainerAndParents

File: src/Controller/Helpers/StructuredDataHelper.php [hunk 2]
```diff
@@ -124,15 +125,16 @@ public function getSchemaForProgrammeContainerAndParents(Programme $programmeCon
...
     }
 
-    public function getSchemaForCollection(Programme $programmeContainer): array
+    public function getSchemaForCollection(Collection $collection, Programme $programme): array
     {
-        $schemaContext = $this->getSchemaForCollectionContainer($programmeContainer);
-        if ($programmeContainer->isTlec()) {
+
+        $schemaContext = $this->schemaHelper->getSchemaForCollection($collection);
+        if ($collection->isTlec() or !$programme) {
```
Comment (BjornTwachtmann) [inline src/Controller/Helpers/StructuredDataHelper.php]: Remove that

**Comment by tomwinnington** [comment]
Test this please

**Pull request merged**